src/main/java/net/sf/jabref/logic/importer/fetcher/GoogleScholar.java

```diff
@@ -62,4 +88,81 @@
 
         return pdfLink;
     }
+
+    @Override
+    public String getName() {
+        return "Google Scholar";
+    }
+
+    @Override
+    public HelpFile getHelpPage() {
+        return HelpFile.FETCHER_GOOGLE_SCHOLAR;
+    }
+
+    @Override
+    public List<BibEntry> performSearch(String query) throws FetcherException {
+        try {
+            obtainAndModifyCookie();
+
+            List<BibEntry> foundEntries = new ArrayList<>(10);
+
+            String queryURL = String.format(BASIC_SEARCH_URL, URLEncoder.encode(query, StandardCharsets.UTF_8.name()));
+            addHitsFromQuery(foundEntries, queryURL);
+            String content;
+
+            if(foundEntries.size()==10) {
+                String secondPage = queryURL+"&start=10";
+                addHitsFromQuery(foundEntries, secondPage);
+            }
+
+            return foundEntries;
+        } catch (IOException e) {
+            throw new FetcherException("Fetching entries from Google Scholar failed: ", e);
+        }
+    }
+
+    private void addHitsFromQuery(List<BibEntry> entryList, String queryURL) throws IOException, FetcherException {
+        String content = URLDownload.createURLDownloadWithBrowserUserAgent(queryURL)
+                .downloadToString(StandardCharsets.UTF_8);
+
+        Matcher matcher = LINK_TO_BIB_PATTERN.matcher(content);
+        while (matcher.find()) {
+            String citationsPageURL = matcher.group().replace("&amp;", "&");
+            BibEntry newEntry = downloadEntry(citationsPageURL);
+            entryList.add(newEntry);
+        }
+    }
+
+    private BibEntry downloadEntry(String link) throws IOException, FetcherException {
+        String downloadedContent = URLDownload.createURLDownloadWithBrowserUserAgent(link).downloadToString(StandardCharsets.UTF_8);
+        BibtexParser parser = new BibtexParser(importFormatPreferences);
+        ParserResult result = parser.parse(new StringReader(downloadedContent));
+        if ((result == null) || (result.getDatabase() == null)) {
+            throw new FetcherException("Parsing entries from Google Scholar bib file failed.");
+        } else {
+            Collection<BibEntry> entries = result.getDatabase().getEntries();
+            if (entries.size() != 1) {
+                LOGGER.debug(entries.size() + " entries found! (" + link + ")");
+                throw new FetcherException("Parsing entries from Google Scholar bib file failed.");
+            } else {
+                BibEntry entry = entries.iterator().next();
+                entry.clearField(BibEntry.KEY_FIELD);
```

Comment:
The other fetcher keep the bibtexkey. Not sure what the best approach is, but we should be consistent.